framework/src/base/MeshGeneratorSystem.C

```diff
@@ -194,6 +195,14 @@ MeshGeneratorSystem::createAddedMeshGenerators()
                              "' does not exist");
   }
 
+  // Check compatibility for CLI / meshing options with csg_only
+  const bool csg_only = getCSGOnly();
+  if (csg_only && _data_driven_generator_name)
+    moose_mesh->paramError(data_driven_generator_param,
+                           "This parameter should not be set in conjunction with --csg-only");
+  if (csg_only && libMesh::n_threads() > 1)
+    mooseWarning("--csg-only option does not currently support multi-threading");
```

Comment:
Even with threads, we don't ever construct threaded copies of MeshGenerators and they're executed outside of threads. I'm not sure this is a problem